File: flow-tests/test-frontend/vite-production/config/templates/templates/PolymerComponent.ts
import {PolymerElement,html} from '@polymer/polymer/polymer-element.js';

class PolymerComponent extends PolymerElement {

  static get template() {
    return html`<span id="label">Default</span>`;
  }
  static get is() {
    return 'polymer-component';
  }
}

customElements.define(PolymerComponent.is, PolymerComponent);

File: flow-tests/test-frontend/vite-test-assets/src/main/java/com/vaadin/viteapp/views/template/PolymerComponent.java
package com.vaadin.viteapp.views.template;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.html.Span;
import com.vaadin.flow.component.polymertemplate.PolymerTemplate;
import com.vaadin.flow.component.template.Id;
import com.vaadin.flow.templatemodel.TemplateModel;

@Tag(PolymerComponent.TAG)
@JsModule("./templates/PolymerComponent.ts")
public class PolymerComponent extends PolymerTemplate<TemplateModel> {
    public static final String TAG = "polymer-component";

    @Id("label")
    private Span label;

    public void setLabel(String value) {
        label.setText(value);
    }
}
